layouts/_default/baseof.html

```diff
@@ -6,8 +6,8 @@
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
 
     {{ with .Site.Params.author }}<meta name="author" content="{{ . }}">{{ end }}
-    {{ with .Site.Params.description }}<meta name="description" content="{{ . }}">{{ end }}
-    {{ with .Site.Params.keywords }}<meta name="keywords" content="{{ . }}">{{ end }}
+    <meta name="description" content="{{ with .Description }}{{ . }}{{ else }}{{ if .IsPage }}{{ with .Params.description }}{{ . }}{{ else }}{{ .Summary }}{{ end }}{{ else }}{{ .Site.Params.description }}{{ end }}{{ end }}">
+    <meta name="keywords" content="{{ with .Keywords }}{{ delimit . "," }}{{ else }}{{ if .IsPage }}{{ with .Params.keywords }}{{ delimit . "," }}{{ else }}{{ .Site.Params.keywords }}{{ end }}{{ else }}{{ .Site.Params.keywords }}{{ end }}{{ end }}">
```

Comment:
I think we can simplify this part using chained `defaults`. Something like this:

```
<meta name="description" content="{{ .Description | default (.Summary | default .Site.Params.description ) }}">
<meta name="keywords" content="{{ (delimit .Keywords ",") | default .Site.Params.keywords }}">
```
```

Reply:
You are right 👍, and I have simplified it now